### PR TITLE
[do not merge] Update RGB --> DEM in style spec and one GL JS example

### PR DESCRIPTION
**Pull request by domlet**
Related to: https://github.com/mapbox/help/issues/2992
Dependent upon: https://github.com/mapbox/help/pull/2999

**Mapbox Terrain-DEM v1 raster tileset** is now live; this is an optimized version of the existing Mapbox Terrain-RGB raster tileset (v0). Whereas the new version offers performance improvements to customers, we should update references to the v0 tileset to use and recommend v1.

- change hillshafe example source from `mapbox://mapbox.terrain-rgb` to `mapbox://mapbox.mapbox-terrain-dem-v1`
- change link from [troubleshooting guide](https://docs.mapbox.com/help/troubleshooting/access-elevation-data/) to [Mapbox data](https://docs.mapbox.com/help/getting-started/mapbox-data/) getting started guide.
- update Style Spec to indicate that `raster-dem` layer type is only compatible with v1.

### CODE REVIEW

File: docs/pages/style-spec/sources.md [hunk 1]
```diff
@@ -142,12 +142,12 @@ https://github.com/mapbox/mapbox-gl-js/blob/main/src/style-spec/reference/v8.jso
...
 
-A raster DEM source. Only supports [Mapbox Terrain RGB](https://blog.mapbox.com/global-elevation-data-6689f1d0ba65) (`mapbox://mapbox.terrain-rgb`):
+A raster DEM source. Only supports [Mapbox Terrain-DEM](https://docs.mapbox.com/help/getting-started/mapbox-data/#mapbox-terrain-dem) (`mapbox://mapbox-terrain-dem-v1`):
 
 ```json
-"mapbox-terrain-rgb": {
+"mapbox-terrain-dem-v1": {
     "type": "raster-dem",
-    "url": "mapbox://mapbox.terrain-rgb"
+    "url": "mapbox://mapbox-terrain-dem-v1"
```
Comment (jakepruitt) [inline docs/pages/style-spec/sources.md:150]: This needs to be `mapbox://mapbox.mapbox-terrain-dem-v1`

**Review by jakepruitt — CHANGES_REQUESTED** [review]

File: docs/pages/style-spec/sources.md (hunk 1 above)
Comment (domlet) [inline docs/pages/style-spec/sources.md:150]:
```suggestion
    "url": "mapbox://mapbox.mapbox-terrain-dem-v1"
```

Comment (domlet) [reply docs/pages/style-spec/sources.md]: wonderful, thanks for catching this @jakepruitt 
corrected in 32f2a76

**Review by jakepruitt — APPROVED** [review]

File: docs/pages/style-spec/sources.md [hunk 2]
```diff
@@ -142,12 +142,12 @@ https://github.com/mapbox/mapbox-gl-js/blob/main/src/style-spec/reference/v8.jso
 
 ## raster-dem
 
-A raster DEM source. Only supports [Mapbox Terrain RGB](https://blog.mapbox.com/global-elevation-data-6689f1d0ba65) (`mapbox://mapbox.terrain-rgb`):
+A raster DEM source. Only supports [Mapbox Terrain-DEM](https://docs.mapbox.com/help/getting-started/mapbox-data/#mapbox-terrain-dem) (`mapbox://mapbox-terrain-dem-v1`):
```
Comment (HeyStenson) [inline docs/pages/style-spec/sources.md:145]:
```suggestion
A raster DEM source. Only supports [Mapbox Terrain-DEM](https://docs.mapbox.com/help/getting-started/mapbox-data/#mapbox-terrain-dem) (`mapbox://mapbox.mapbox-terrain-dem-v1`):
```

❓ 🤔

**Pull request merged**